daemon/endpoint.go

```diff
@@ -211,6 +211,44 @@ func (h *putEndpointID) Handle(params PutEndpointIDParams) middleware.Responder
 	if err != nil {
 		apierror.Error(code, err)
 	}
+
+	// Wait for endpoint to be ready if specified in API call.
+	if params.Endpoint.SyncBuildEndpoint {
+		log.Debug("synchronously waiting for endpoint to be in %s state", endpoint.StateReady)
+
+		e, err := endpointmanager.Lookup(params.ID)
+		if err != nil {
+			return apierror.Error(PutEndpointIDFailedCode, err)
```

Comment:
We have to remove the endpoint again this case. The API must be consistent and either return success with the endpoint still inserted or an error without leaving behind an endpoint.

Reply:
Done.